R/ab_scenario.R

```diff
@@ -18,122 +20,90 @@
-#' ab_evening_dutch = ab_scenario(
-#'   leeds_houses,
-#'   leeds_buildings,
-#'   leeds_desire_lines,
-#'   leeds_zones,
-#'   scenario = "dutch",
-#'   output_format = "json_list",
-#'   hr = 20, # representing 8 pm
-#'   sd = 0
-#' )
-#' f = tempfile(fileext = ".json")
-#' ab_save(ab_evening_dutch, f)
-#' readLines(f)[13]
-#' 20 * 60^2
-ab_scenario = function(houses,
-                       buildings,
-                       desire_lines,
-                       zones,
-                       scenario = "base",
-                       time_fun = ab_time_normal,
-                       output_format = "sf",
-                       op = sf::st_intersects,
-                       ...
-                       ) {
-
-  requireNamespace("sf", quietly = TRUE)
-  # loop over each desire line
-  i = 1
-  for(i in seq(nrow(desire_lines))) {
-    # todo: allow the user to choose the 'pop var' or document (RL 2020-02-10)
-    pop = desire_lines$all_base[i]
-    cnames = names(desire_lines)
-
-    # todo: split out the lines to odc_to_sf as function? (RL 2020-02-10)
-    # supressing messages associated with GEOS ops on unprojected data
-    suppressWarnings({
-      suppressMessages({
-        origins = houses[sample(nrow(houses), size = pop, replace = TRUE), ]
-        destination_zone = zones %>% subset(geo_code == desire_lines$geo_code2[i])
-        destination_buildings = buildings[destination_zone, , op = op]
-        destinations = destination_buildings[sample(nrow(destination_buildings), size = pop, replace = TRUE), ]
-        origin_coords = origins %>% sf::st_centroid() %>% sf::st_coordinates()
-        destination_coords = destinations %>% sf::st_centroid() %>% sf::st_coordinates()
-        desire_lines_disag = od::odc_to_sf(odc = cbind(origin_coords, destination_coords))
-      })
+#' plot(ablines["mode"], add = TRUE)
+#' ablines_json = ab_json(ablines)
+#' # ablines = ab_scenario(
+#' #   leeds_houses,
+#' #   leeds_buildings,
+#' #   leeds_desire_lines,
+#' #   leeds_zones,
+#' #   output_format = "sf"
+#' # )
+#' # dutch = ab_scenario(
+#' #   leeds_houses,
+#' #   leeds_buildings,
+#' #   leeds_desire_lines,
+#' #   leeds_zones,
+#' #   scenario = "dutch",
+#' #   output_format = "sf"
+#' # )
+#' # plot(ablines, key.pos = 1, reset = FALSE)
+#' # plot(leeds_site_area$geometry, add = TRUE)
+#' # plot(leeds_buildings$geometry, add = TRUE)
+#' # plot(dutch, key.pos = 1, reset = FALSE)
+#' # plot(leeds_site_area$geometry, add = TRUE)
+#' # plot(leeds_buildings$geometry, add = TRUE)
+#' # dutch$departure = ab_time_normal(hr = 8, sd = 0.5, n = nrow(dutch))
+#' # ab_evening_dutch = ab_scenario(
+#' #   leeds_houses,
+#' #   leeds_buildings,
+#' #   leeds_desire_lines,
+#' #   leeds_zones,
+#' #   scenario = "dutch",
+#' #   output_format = "json_list",
+#' #   hr = 20, # representing 8 pm
+#' #   sd = 0
+#' # )
+#' # f = tempfile(fileext = ".json")
+#' # ab_save(ab_evening_dutch, f)
+#' # readLines(f)[13]
+#' # 20 * 60^2
```

Comment:
I think it's just my unfamiliarity with R, but this looks like example code for how to use this function. What's the meaning/use of this bit of math that I assume means 20 hours?

Reply:
It was just there to highlight the fact that the resulting departure times are not to dissimilar to 20:00 represented in seconds past midnight, which is the unit of time in `abstr` I believe. Commented out as it was a rather ActDev specific bit and we could remove it - same with the other commented-out examples.